### DISCUSSION TITLE
Federation and Node interface

### DISCUSSION THREAD

**Discussion by theprobugmaker** [99designs/gqlgen#1968] [Q&A]
### What happened?

I'm trying to use the Relay Connections specification for pagination with gqlgen + entgo (from this tutorial: https://entgo.io/docs/tutorial-todo-gql-node) but when I create an `interface Node` that the type `User` implements, I'm getting an error.

> merging type systems failed: unable to bind to interface: poc/ent.User does not satisfy the interface github.com/99designs/gqlgen/plugin/federation/fedruntime.Entity
> exit status 1

### What did you expect?

I need the `Node` type to work with the Relay Connections specification.

### Minimal graphql.schema and models to reproduce

```graphql
interface Node {
  id: ID!
}

type User implements Node @key(fields: "id") {
  id: ID!
}
```

### versions
 - `gqlgen version`? 0.16.0
 - `go version` 1.17

**Comment by eleretic** [comment]
You have to create a function called `IsNode` on your User model
If you need to resolve entitites with federation, you need to add `IsEntity` as well
```
func (User) IsEntity() {}
func (User) IsNode() {}
```
